src/pudl/workspace/datastore.py

```diff
@@ -56,7 +57,15 @@ def get_resource_path(self, name: str) -> str:
         res = self._get_resource_metadata(name)
         # remote_url is sometimes set on the local cached version of datapackage.json
         # so we should be using that if it exists.
-        return res.get("remote_url") or res.get("path")
+        resource_path = res.get("remote_url") or res.get("path")
+        parsed_path = urlparse(resource_path)
+        if parsed_path.path.startswith("/api/files"):
+            record_number = self.doi.lower().rsplit("zenodo.", 1)[-1]
+            new_path = f"/api/records/{record_number}/files/{name}/content"
+            new_url = ParseResult(**(parsed_path._asdict() | {"path": new_path}))
```

Comment:
See the response from Zenodo support about what the best pseudo-persistent URL to use is.  His suggestion does seem like the simplest possible URL, without attributes that seem dependent on any API specifics...  here it would be:

```py
new_path = f"records/{record_number}/files/{name}"
```

Given that we're not **really** using the API here and are instead constructing the URL we would expect to receive from the API, does it make sense to go with the non-API URL support suggested? Or is there a good reason to go with the expected API URL?

Reply:
Pretending that we used this most-stable URL all along seems like a fine idea!